sdk/core/core-client-rest/src/multipart.ts

```diff
@@ -0,0 +1,186 @@
+
+/**
+ * Describes a single part in a multipart body.
+ */
+export interface PartDescriptor {
+  /**
+   * Content type of this part. If set, this value will be used to set the Content-Type MIME header for this part, although explicitly
+   * setting the Content-Type header in the headers bag will override this value. If set to `null`, no content type will be inferred from
+   * the body field. Otherwise, the value of the Content-Type MIME header will be inferred based on the type of the body.
+   */
+  contentType?: string | null;
+
+  /**
+   * The disposition type of this part (for example, "form-data" for parts making up a multipart/form-data request). If set, this value
+   * will be used to set the Content-Disposition MIME header for this part, in addition to the `name` and `filename` properties.
+   * Explicitly setting the Content-Disposition header in the headers bag will override this value.
+   */
+  dispositionType?: string;
+
+  /**
+   * The field name associated with this part. This value will be appended to the Content-Disposition header if dispositionType is set.
+   */
+  name?: string;
+
+  /**
+   * The file name of the content if it is a file. This will be appended to the Content-Disposition header value if dispositionType is set.
+   */
+  filename?: string;
+
+  /**
+   * The multipart headers for this part of the multipart body. Values of the Content-Type and Content-Disposition headers set in the headers bag
+   * will take precedence over those computed from the request body or the contentType, dispositionType, name, and filename fields on this object.
+   */
+  headers?: Record<string, string>;
+
+  /**
+   * The body of this part of the multipart request.
+   */
+  body?: unknown;
+}
+
+type MultipartBodyType = BodyPart["body"];
+
+/**
+ * Get value of a header in the part descriptor ignoring case
+ */
+function getHeaderValue(descriptor: PartDescriptor, headerName: string): string | undefined {
+  if (descriptor.headers) {
+    const actualHeaderName = Object.keys(descriptor.headers).find(
+      (x) => x.toLowerCase() === headerName.toLowerCase(),
+    );
+    if (actualHeaderName) {
+      return descriptor.headers[actualHeaderName];
+    }
+  }
+
+  return undefined;
+}
+
+function getPartContentType(descriptor: PartDescriptor): string | undefined {
+  const contentTypeHeader = getHeaderValue(descriptor, "content-type");
+  if (contentTypeHeader) {
+    return contentTypeHeader;
+  }
+
+  // Special value of null means content type is to be omitted
+  if (descriptor.contentType === null) {
+    return undefined;
+  }
+
+  if (descriptor.contentType) {
+    return descriptor.contentType;
+  }
+
+  const { body } = descriptor;
+
+  if (body === null || body === undefined) {
+    return undefined;
+  }
+
+  if (typeof body === "string" || typeof body === "number" || typeof body === "boolean") {
+    return "text/plain; charset=UTF-8";
+  }
+
+  if (body instanceof Blob) {
```

Comment:
will we trust the type defined in File also?
```ts
new File([data],"a.txt", { type: "text/plain" })
```

Reply:
`File` inherits from `Blob` so this check handles the `File` case as well.